### ISSUE TITLE
[Feature]: Set custom artifact name in WandbModelCheckpoint

### ISSUE THREAD

**Issue by GabrielGosden** [wandb/wandb#8834]
### Description

Hi,

I've noticed that when using [WandbModelCheckpoint's](https://docs.wandb.ai/ref/python/integrations/keras/wandbmodelcheckpoint/) it's currently not possible to define a custom name for the model artifact. The name is automatically set to  `run_{wandb.run.id}_model` as specified in the [_log_ckpt_as_artifact](https://github.com/wandb/wandb/blob/6bc2440183e02633c11fdd291550c5eb9c0b4634/wandb/integration/keras/callbacks/model_checkpoint.py#L154) function.

This presents a challenge when saving models for multiple metrics within a single run. For instance, consider having a WandbModelCheckpoint for different metrics such as loss, recall, precision, F1 score, IoU, Dice, etc. Since the artifact name remains the same across all metrics, the model versioning becomes ineffective. This results in every model for a specific metric being stored under the same artifact name, leading to many versions that aren't metric-specific.

Consequently, identifying the best-performing model for a particular metric can become quite cumbersome. It often requires sifting through log files or exploring the latest model versions.

Here's an example of what the end result looks like:
![Image](https://github.com/user-attachments/assets/b539506d-028a-4e6a-9cb3-198570423d47)

### Suggested Solution

I would like to be able to define the models artifact name when specifying the callbacks for [WandbModelCheckpoint's](https://docs.wandb.ai/ref/python/integrations/keras/wandbmodelcheckpoint/).

**Comment by GabrielGosden** [comment]
Slightly related issues:

https://github.com/wandb/wandb/issues/7413
https://github.com/wandb/wandb/issues/6278

People are having the same issue with `run.log_artifact`

**Comment by paulosabile-wb** [comment]
Hey @GabrielGosden Good day and thank you for writing in!

Happy to review this for you! I would like to know the context for this so we can review and identify the next steps to suggest. Would you mind sharing a code snippet so we can review how you are currently executing your callbacks for `WandbModelCheckpoint`? We will then analyze your current setup and discuss this with the team.

It would also help if you can share your current SDK version. You can get this by running `wandb --version`. Thanks!

**Comment by GabrielGosden** [comment]
Hey @paulosabile-wb,

I have created a minimal example using `wandb==0.19.0` 

This minimal example creates a `WandbModelCheckpoint` for `loss`,  `val_loss`, `accuracy` and `val_accuracy`.  This could be recall, precision, F1 score, IoU, Dice, etc.

```
import wandb
from wandb.integration.keras import WandbModelCheckpoint
import os 
import numpy as np
import tensorflow as tf

wandb.init(project="feature_request_model_checkpoint")

# Create a simple model
model = tf.keras.Sequential([
    tf.keras.layers.Dense(10, input_dim=10, activation='relu'),
    tf.keras.layers.Dense(1, activation='linear')
])

# Compile arbitrary model.
model.compile(optimizer='adam',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

# Define multiple WandbModelCheckpoint callbacks
callbacks = [
    WandbModelCheckpoint(
        os.path.join("models/", wandb.run.name + "_min_loss.keras"),
        monitor="loss",
        verbose=1,
        mode="min",
        save_best_only=False,
        save_weights_only=False,
    ),
    WandbModelCheckpoint(
        os.path.join("models/", wandb.run.name + "_min_val_loss.keras"),
        monitor="val_loss",
        verbose=1,
        mode="min",
        save_best_only=False,
        save_weights_only=False,
    ),
    WandbModelCheckpoint(
        os.path.join("models/", wandb.run.name + "max_accuracy.keras"),
        monitor="accuracy",
        verbose=1,
        mode="max",
        save_best_only=False,
        save_weights_only=False,
    ),
    WandbModelCheckpoint(
        os.path.join("models/", wandb.run.name + "max_val_accuracy.keras"),
        monitor="val_accuracy",
        verbose=1,
        mode="min",
        save_best_only=False,
        save_weights_only=False,
    )
]

# Train the model
model.fit(
    x=np.random.rand(100, 10),
    y=np.random.rand(100, 1),
    epochs=100,
    validation_data=(np.random.rand(20, 10), np.random.rand(20, 1)),
    callbacks=callbacks
)
```
This creates >100 model artifacts all with the same name. Finding the best performing model on a given metric is now quite difficult.

![Image](https://github.com/user-attachments/assets/7cef8284-d943-4f22-bc2d-5b007b94d773)

**Comment by paulosabile-wb** [comment]
Thanks @GabrielGosden Let me check this for a while and I will get back to you.

**Comment by paulosabile-wb** [comment]
Hi @GabrielGosden Good day! I have logged a feature request for this. We will keep you posted for any updates. Thank you for raising this to us!

**Comment by GabrielGosden** [comment]
@paulosabile-wb Are there any updates on this feature request?

**Comment by paulosabile-wb** [comment]
Hi @GabrielGosden Good day and thank you for your follow up. We currently do not have an ETA yet for the feature request but we'll make sure to keep you posted for updates once we receive feedback or details about the progress. We thank you for your patience.
